File: plugins/git-utils/src/commands/reviews.ts
// ============================================================
// reviews command (← unresolved-reviews.sh)
// ============================================================
// CLI: bun run src/cli.ts reviews [pr-number]
//
// 동작:
//   1. PR 번호 결정 (인자 or 현재 브랜치에서 자동 감지)
//   2. gh api graphql로 리뷰 쓰레드 조회
//   3. JSON 출력
//
// 기존 unresolved-reviews.sh 대비 개선:
//   - GraphQL 쿼리를 TypeScript 내 관리
//   - GitHubService 주입으로 테스트 가능
// ============================================================

import type { Command, ReviewsInput, ReviewsOutput } from '../types';
import type { GitHubService } from '../core';

export interface ReviewsDeps {
  github: GitHubService;
}

export type ReviewsCommand = Command<ReviewsInput, ReviewsOutput>;

File: plugins/atelier/cli/src/git/commands/reviews.rs
//! `reviews` command, ported from git-utils `commands/reviews.ts`
//! (originally `unresolved-reviews.sh`). Resolves the PR number (explicit or
//! detected) and fetches its review threads.

use crate::git::core::github::GitHubService;
use crate::git::types::{ReviewsInput, ReviewsOutput};

/// The `reviews` command, backed by an injected [`GitHubService`].
pub struct ReviewsCommand<'a> {
    github: &'a dyn GitHubService,
}

impl<'a> ReviewsCommand<'a> {
    pub fn new(github: &'a dyn GitHubService) -> Self {
        Self { github }
    }

    pub fn run(&self, input: &ReviewsInput) -> Result<ReviewsOutput, String> {
        let pr_number = match input.pr_number {
            Some(n) => n,
            None => match self
                .github
                .detect_current_pr_number()
                .map_err(|e| e.to_string())?
            {
                Some(n) => n,
                None => {
                    return Err(
                        "No PR found. Provide a PR number or checkout a PR branch.".to_string()
                    )
                }
            },
        };

        self.github
            .get_review_threads(pr_number)
            .map_err(|e| e.to_string())
    }
}
